src/stock-orderable-group/orderable-group.service.spec.js

```diff
@@ -0,0 +1,299 @@
+        });
+
+        this.lot1 = {
+            id: 'lot id 1',
+            expirationDate: '2022-05-08'
+        };
+        this.lot2 = {
+            id: 'lot id 2',
+            expirationDate: '2019-01-20'
+        };
+        this.lot3 = {
+            id: 'lot id 3',
+            expirationDate: '2018-04-03'
+        };
+
+        this.item1 = {
+            orderable: {
+                id: 'a'
+            },
+            lot: this.lot1
+        };
+        this.item2 = {
+            orderable: {
+                id: 'a'
+            }
+        };
+        this.item3 = {
+            orderable: {
+                id: 'b'
+            }
+        };
+        this.item4 = {
+            orderable: {
+                id: 'a'
+            },
+            lot: this.lot2
+        };
+        this.item5 = {
+            orderable: {
+                id: 'a'
+            },
+            lot: this.lot3
+        };
+        this.items = [this.item1, this.item2, this.item3];
+
+        this.kitConstituents = [
+            new this.OrderableChildrenDataBuilder().withId('child_product_1_id')
+                .withQuantity(30)
+                .buildJson()
+        ];
+        this.orderable = new this.OrderableDataBuilder().withChildren(this.kitConstituents)
+            .buildJson();
+        this.kitOrderableGroup = new this.OrderableGroupDataBuilder().withOrderable(this.orderable)
+            .build();
+        this.orderableGroups = [
+            new this.OrderableGroupDataBuilder().withOrderable(
+                new this.OrderableDataBuilder().withChildren([])
+                    .buildJson()
+            )
+                .build(),
+            new this.OrderableGroupDataBuilder().withOrderable(this.orderable)
+                .build()
+        ];
+
+    });
+
+    it('should group items by orderable id', function() {
+        expect(this.orderableGroupService.groupByOrderableId(this.items)).toEqual([
+            [this.item1, this.item2],
+            [this.item3]
+        ]);
+    });
+
+    it('should find item in group by lot', function() {
+        expect(this.orderableGroupService.findByLotInOrderableGroup(this.items, this.lot1)).toBe(this.item1);
+    });
+
+    it('should find item in group by NULL lot', function() {
+        expect(this.orderableGroupService.findByLotInOrderableGroup(this.items, null)).toBe(this.item2);
+    });
+
+    it('should find item with new lot', function() {
+        var newLot = new this.LotDataBuilder().build(),
+            newItem = this.item2;
+        newItem.lot = newLot;
+        newItem.stockOnHand = 0;
+
+        expect(this.orderableGroupService.findByLotInOrderableGroup(this.items, newLot)).toBe(newItem);
+    });
+
+    it('should find lots in orderable group', function() {
+        //given
+        var group = [this.item1, this.item2];
+
+        //when
+        var lots = service.lotsOf(group);
+
+        // ANGOLASUP-516: Removed the 'No Lot Defined' option
```

Comment:
same here

Reply:
It is related to the 'removed' code, not 'added' one (as we discussed on slack)